alacritty_terminal/src/term/mod.rs

```diff
@@ -944,8 +1214,10 @@ impl<T: EventListener> Handler for Term<T> {
             (Line(0), self.bottommost_line())
         };
 
-        self.grid.cursor.point.line = max(min(line + y_offset, max_y), Line(0));
-        self.grid.cursor.point.column = min(col, self.last_column());
+        self.damage_cursor();
+        self.grid.cursor.point.line = cmp::max(cmp::min(line + y_offset, max_y), Line(0));
+        self.grid.cursor.point.column = cmp::min(col, self.last_column());
+        self.damage_cursor();
```

Comment:
Damaging cursor twice here seems inefficient? Shouldn't the `damage_cursor` function automatically compute a delta to its last position?

Reply:
You change the line here, so you should damage old and new one. We could try moving `damage.last_cursor` around though.

Reply:
Moving last_cursor around gets complicated when you start moving it back and forth. I think I'd rather do simpler, but in a way that you can't really fail with collecting damage.